inst/include/dust/random/nbinomial.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef DUST_RANDOM_NBINOMIAL_HPP
+#define DUST_RANDOM_NBINOMIAL_HPP
+
+#include <cmath>
+
+#include "dust/random/gamma.hpp"
+#include "dust/random/poisson.hpp"
+#include "dust/random/generator.hpp"
+
+namespace dust {
+namespace random {
+
+namespace {
+
+template <typename real_type>
+void nbinomial_validate(real_type size, real_type prob) {
+   if(!R_FINITE(size) || !R_FINITE(prob) || size <= 0 || prob <= 0 || prob > 1) {
+    char buffer[256];
+    snprintf(buffer, 256,
+             "Invalid call to nbinomial with size = %g, prob = %g",
+             size, prob);
+    dust::utils::fatal_error(buffer);
+  }
+}
+
+}
+
+template <typename real_type, typename rng_state_type>
+real_type nbinomial(rng_state_type& rng_state, real_type size, real_type prob)
+{
+    nbinomial_validate(size, prob);
```

Comment:
Can you add an assert that this does not work for CUDA, and a deterministic route (as in the gamma case). If you could move that closing brace too please - probably we should set up clang-format or something at some point